Source/CsDebugScript.Engine/CodeArray.cs

```diff
@@ -71,7 +71,7 @@ public CodeArray(Variable variable)
         /// <param name="length">The array length.</param>
         public CodeArray(Variable variable, int length)
         {
-            if (!variable.GetCodeType().IsArray && !variable.GetCodeType().IsPointer)
+            if (!variable.GetCodeType().IsArray && !variable.GetCodeType().IsPointer && length > 0)
```

Comment:
Please move "&& length > 0" to new if statement and throw different exception. Wrong code type is not describing it correctly.